analyzer/codechecker_analyzer/analyzers/clangtidy/analyzer.py

```diff
@@ -227,16 +229,30 @@ def get_checker_list(self, config) -> Tuple[List[str], List[str]]:
             state, _ = value
 
             # Checker name is a compiler warning.
+            if checker_name.startswith('W'):
+                LOG.warning(f"The following usage of {checker_name} "
+                            "compiler warning as a checker name is "
+                            "deprecated, please use "
+                            "clang-diagnostic-<checker-name> instead")
+
             warning_name = get_compiler_warning_name(checker_name)
             if warning_name is not None:
+                # -W and clang-diagnostic- are added as compiler warnings.
                 if state == CheckerState.enabled:
-                    compiler_warnings.append('-W' + warning_name)
+                    compiler_warnings.add('-W' + warning_name)
                 elif state == CheckerState.disabled:
-                    compiler_warnings.append('-Wno-' + warning_name)
-
+                    compiler_warnings.add('-Wno-' + warning_name)
                 continue
-            elif checker_name.startswith('clang-diagnostic'):
-                checker_name += '*'
+
+            # If clang-diagnostic is enabled, else add it as a disabled
+            # check.
+            if checker_name.startswith('clang-diagnostic-') and \
```

Comment:
We already have the variable `warning_name`, that holds the information on whether we recognized the checker as compiler warning, no? Why are we reinveting the wheel instead of reusing the result of `get_compiler_warning_name`?

Reply:
I still think this is bad design. The code would look more readable if we moved that logic to `get_compiler_warning_name` (where it belongs), and we really need to, check whether this is a disabled clang-diagnostic _after the fact_.